src/components/AddPerson.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useState, ChangeEvent } from 'react';
+import { AddPersonInputs } from './AddPersonInputs';
+import { AddPersonButton } from './AddPersonButton';
+import { SexSelect } from './SexParametrs';
+import { AddFather } from './AddFather';
+import { AddMother } from './AddMother';
+import { Redirect } from 'react-router-dom'
+
+const pattern = /[0-9]| {2}|\!|\.|\@|\#|\$|\%|\^|\&|\*|\(|\)|^\s+/g;
+const currentYear = new Date().getFullYear();
+
+interface Props {
+  people: People[];
+  addPerson: (
+    name: string,
+    born: string,
+    died: string,
+    sex: string,
+    father: string,
+    mother: string,
+  ) => void;
+}
+
+export const AddPerson: React.FC<Props> = ({ people, addPerson }) => {
+  const [nameValue, setNameValue] = useState('');
+  const [bornValue, setBornValue] = useState('');
+  const [diedValue, setDiedValue] = useState('');
+  const [gender, setGender] = useState('');
+  const [father, setFather] = useState('');
+  const [mother, setMother] = useState('');
+  const [isAdd, setIsAdd] = useState(false);
+  const [errorMother, setErrorMother] = useState(false);
+  const [errorFather, setErrorFather] = useState(false);
+  const [errorName, setErrorName] = useState(false);
+  const [errorBorn, setErrorBorn] = useState(false);
+  const [errorDied, setErrorDied] = useState(false);
+  const [minusDifference, setMinusDifference] = useState(false);
+  const [bigDifference, setBigDifference] = useState(false);
+  const [errorGender, setErrorGender] = useState(false);
```

Comment:
```suggestion
  const [errors, setErrors] = useState({
    errorFather: false,
    errorMother: false,
    errorName: false,
    ....
  });
```
Дальше там где надо вызываешь 
setErrors(prevErrors => ({ ...prevErrors, errorFather: true }));

Reply:
Возможно чуть позже именно так и сделаю